homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,188 @@
+    CONF_AUTHENTICATION,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+)
+
+from .camera import DEFAULT_CONTENT_TYPE, GenericCamera
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_RTSP_TRANSPORT,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    DEFAULT_NAME,
+    DOMAIN,
+    FFMPEG_OPTION_MAP,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DATA = {
+    CONF_NAME: DEFAULT_NAME,
+    CONF_AUTHENTICATION: HTTP_BASIC_AUTHENTICATION,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE: False,
+    CONF_CONTENT_TYPE: DEFAULT_CONTENT_TYPE,
+    CONF_FRAMERATE: 2,
+    CONF_VERIFY_SSL: True,
+}
+
+
+class GenericIPCamConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for generic IP camera."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def _test_connection(self, info) -> tuple[bool, str]:
+        """Verify that the camera data is valid before we add it."""
+        unique_id = self.flow_id
+        if unique_id in self._async_current_ids():
+            return False, "unique_id_already_in_use"
+        await self.async_set_unique_id(unique_id)
+        cam = GenericCamera(self.hass, info, unique_id)
+        # Minimum functionality is getting a still image
+        image = await cam.async_camera_image()
+        if image is None:
+            return False, "unable_still_load"
+        fmt = imghdr.what(None, h=image)
+        _LOGGER.debug(
+            "Still image at '%s' detected format: %s", info[CONF_STILL_IMAGE_URL], fmt
+        )
+        if fmt is None:
+            return False, "invalid_still_image"
+
+        # Second level functionality is to get a stream.
+        if info.get(CONF_STREAM_SOURCE):
+            try:
+                stream_source = info[CONF_STREAM_SOURCE]
+                # For RTSP streams, prefer TCP. This code is duplicated from
+                # homeassistant.components.stream.__init__.py:create_stream()
+                # It may be possible & better to call create_stream() directly.
+                stream_options: dict[str, str]
+                if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
+                    stream_options = {
+                        "rtsp_flags": "prefer_tcp",
+                        "stimeout": "5000000",
+                    }
+                else:
+                    stream_options = {}
+                if info.get(CONF_RTSP_TRANSPORT):
+                    stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = info[
+                        CONF_RTSP_TRANSPORT
+                    ]
+                container = await self.hass.async_add_executor_job(
+                    partial(
+                        av.open,
+                        info[CONF_STREAM_SOURCE],
```

Comment:
```suggestion
                        stream_source,
```
I think this is doable here because you are still in your if block

Reply:
Thanks! Done